spirv_glsl.cpp

```diff
@@ -7892,18 +7906,45 @@ void CompilerGLSL::emit_glsl_op(uint32_t result_type, uint32_t id, uint32_t eop,
 		else
 			emit_trinary_func_op(result_type, id, args[0], args[1], args[2], "fma");
 		break;
+
 	case GLSLstd450Modf:
 		register_call_out_argument(args[1]);
-		forced_temporaries.insert(id);
-		emit_binary_func_op(result_type, id, args[0], args[1], "modf");
+		if (!is_legacy())
+		{
+			forced_temporaries.insert(id);
+			emit_binary_func_op(result_type, id, args[0], args[1], "modf");
+		}
+		else
+		{
+			//NB. legacy GLSL doesn't have trunc() either, so we do a value cast
+			auto &op1_type = expression_type(args[1]);
+			auto via_type = op1_type;
+			via_type.basetype = SPIRType::Int;
+			statement(to_expression(args[1], false), " = ",
```

Comment:
What you want here is emit_uninitialized_temporary_expression and then you can add statements to fill in the temporary as needed.

Reply:
This doesn't seem to work.  I get an exception "Overwriting a variant with new type."

I tried:
```cpp
emit_uninitialized_temporary_expression(op1_type.self, args[1]);
```

Note that this is an out argument (via OpVariable), not a result.  I assumed `register_call_out_argument` was handling this case, and it seemed to be working.

Reply:
Right, this is targeting args[1]. I was a bit confused since you did to_expression(foo, false). What was the purpose of the "false" here?

Reply:
Just saw your reply.

I added the "false" because while I was trying to figure out why it was always emitting the result temporary, I looked at the argument, it said "register read", thought "well it's not a read", then set it in the hope it'd fix that.  But it didn't.  I'll remove it.